src/Agent.Worker/StepsRunner.cs

```diff
@@ -20,19 +20,18 @@ public interface IStep
         IExecutionContext ExecutionContext { get; set; }
         // Always runs. Even if a previous critical step failed.
         bool Finally { get; }
-        TaskResult? Result { get; set; }
-        Task<TaskResult> RunAsync();
+        Task RunAsync();
```

Comment:
Again, removed the return type (TaskResult). It already lives on every execution context.